components/common-go/process/process.go

```diff
@@ -14,3 +21,36 @@ func IsNotChildProcess(err error) bool {
 
 	return (err.Error() == "wait: no child processes" || err.Error() == "waitid: no child processes")
 }
+
+var ErrForceKilled = errors.New("process didn't temrinate, so we send SIGKILL")
+
+// TerminateSync sends a SIGTERM to the given process and returns when the process has terminated but at most the given gracePeriod
+// if the gracePeriod ends we send SIGKILL to the process and return immediately with ErrForceKilled
+func TerminateSync[T any](pid int, gracePeriodOver <-chan T) error {
```

Comment:
Is it acceptable to go terminate the process even if at the start of this method the channel already has a value?

If not, you can do
```
select {
  case <- gracePeriodOver:
      return ..
   default:
}
```
Which will check the channel, but if no value is available it will continue

Reply:
Thanks, we do want run through the code including the effect of SIGKILL at the end.